manager.go

```diff
@@ -1541,6 +1561,12 @@ func (m *Manager) UpdateActivatedProbes(selectors []ProbesSelector) error {
 			}
 			probe.Enabled = true
 		}
+
+		if probeIsExcluded(probe, m.options.ExcludedFunctions) {
```

Comment:
This is the point where I'm unsure of what I'm doing. If I don't do that this will try to init/attach in the next block (line 1547) and will nil-ptr panic because `probe.programSpec` is nil (of course it was excluded)

Reply:
I don't think excluded functions should have associated probes, so that sounds like a separate bug we should fix. WDYT?

Reply:
I think that's a good idea. I believe the way we currently use this is that we never pass excluded functions in the `UpdateAvailableProbes`, meaning that we never hit this case. But for sure it would be even better to remove excluded probes directly

Reply:
https://github.com/DataDog/ebpf-manager/pull/123

Reply:
Thanks, updated this PR to take this into account